src/utils.py

```diff
@@ -440,3 +473,14 @@ def merge_with_checkpoint(run_args, checkpoint_args):
     run_args.data.sample_rate = checkpoint_args.data.sample_rate
     OmegaConf.set_struct(run_args, True)
     return run_args
+
+
+def flatten(d, parent_key='', sep='.'):
```

Comment:
Do we also need a reverse method (unflatten) to turn back to yaml?
Just thinking out loud

Reply:
I tend to agree, useful for actually running with past configurations. 
Among other things it would mean that we can't use . as a default separator. Note sure what we can use because nearly other char has a use in python (except , which is not elegant). Ideas?

Reply:
@mosheman5 convinced me this is not needed
What we do need is a more fool proof approach to uploading experiments to s3